.evergreen/find-python3.sh

```diff
@@ -0,0 +1,232 @@
+  set -o errexit
+  set -o nounset
+  set -o pipefail
+
+  local -r bin="${1:?'is_venv_capable requires a name or path of a python binary to test'}"
+
+  # Use a temporary directory to avoid polluting the caller's enviornment.
+  local -r tmp="$(mktemp -d)"
+  trap 'rm -rf "$tmp"' EXIT
+
+  # Evaluate the result of this function.
+  "$bin" -m venv "$tmp" 1>&2
+) 1>&2
+
+# is_virtualenv_capable
+#
+# Usage:
+#   is_virtualenv_capable python
+#   is_virtualenv_capable /path/to/python
+#
+# Parameters:
+#   "$1": The name or path of the python binary to test.
+#
+# Return 0 (true) if the given argument $1 can successfully evaluate the
+# command:
+#   "$1" -m virtualenv -p "$1" venv
+# Return a non-zero value (false) otherwise.
+#
+# Diagnostic messages may be printed to stderr (pipe 2). Redirect to /dev/null
+# to silence these messages.
+is_virtualenv_capable() (
+  set -o errexit
+  set -o nounset
+  set -o pipefail
+
+  local -r bin="${1:?'is_virtualenv_capable requires a name or path of a python binary to test'}"
+
+  # Use a temporary directory to avoid polluting the caller's enviornment.
+  local -r tmp="$(mktemp -d)"
+  trap 'rm -rf "$tmp"' EXIT
+
+  # Evaluate the result of this function.
+  "$bin" -m virtualenv -p "$bin" "$tmp" 1>&2
+) 1>&2
+
+# find_python3
+#
+# Usage:
+#   find_python3
+#   PYTHON_BINARY=$(find_python3)
+#   PYTHON_BINARY=$(find_python3 2>/dev/null)
+#
+# Return 0 (true) if a Python 3 binary capable of creating a virtual environment
+# with either venv or virtualenv can be found.
+# Return a non-zero (false) value otherwise.
+#
+# If successful, print the name of the binary stdout (pipe 1).
+# Otherwise, no output is printed to stdout (pipe 1).
+#
+# Diagnostic messages may be printed to stderr (pipe 2). Redirect to /dev/null
+# with `2>/dev/null` to silence these messages.
+#
+# Example:
+#   PYTHON_BINARY=$(find_python3)
+#   if [[ -z "$PYTHON_BINARY" ]]; then
+#     # Handle missing Python binary situation.
+#   fi
+#
+#   if "$PYTHON_BINARY" -m venv -h; then
+#     "$PYTHON_BINARY" -m venv venv
+#   else
+#     "$PYTHON_BINARY" -m virtualenv -p "$PYTHON_BINARY" venv
+#   fi
+find_python3() (
+  set -o errexit
+  set -o nounset
+  set -o pipefail
+
+  local -a bins=()
+  local bin=""
+
+  # The list of Python binaries to test for venv or virtualenv support.
+  # The binaries are tested in the order of their position in the array.
+  {
+    append_bins() {
+      local -r path="${1:?'missing path'}"
+      shift
+      local -r pattern="${1:?'missing pattern'}"
+      shift
+      local -ar suffixes=("${@:?'missing suffixes'}")
+
+      for dir in $(find "$path" -maxdepth 1 -name "$pattern" -type d 2>/dev/null | sort -rV); do
+        for bin in "${suffixes[@]}"; do
+          if is_python3 "$dir/$bin"; then
+            bins+=("$dir/$bin")
+          fi
+        done
+      done
+    }
+
+    # /opt/mongodbtoolchain/vX/bin/python
+    append_bins "/opt/mongodbtoolchain" "v[0-9]*" "bin/python3" "bin/python"
```

Comment:
This should be reconsidered.  We use the toolchain in various places and those uses have the potential to become problematic.  Specifically, the toolchain is intended for the server project, so it can break without warning.  It may be that the way this script checks for presence and capability of Python is resilient against unexpected changes to the toolchain.  If that's the case then a comment to that effect might be good here.